File: src/data/scripts/campaign/econ/industries/niko_MPC_defenseSatelliteLuddicSuppressor.java
package data.scripts.campaign.econ.industries;

import com.fs.starfarer.api.impl.campaign.econ.impl.BaseIndustry;

public class niko_MPC_defenseSatelliteLuddicSuppressor extends BaseIndustry {

    @Override
    public void apply() {
        super.apply(true);
    }

    @Override
    public void unapply() {
        super.unapply();
    }

    @Override
    public float getPatherInterest() {
        return -3f;
    }

    @Override
    public boolean isHidden() { //hidden structures dont block structure construction
        return true;
    }

    @Override
    public boolean isAvailableToBuild() {
        return false;
    }

    @Override
    public boolean showWhenUnavailable() {
        return false;
    }
}

File: src/data/scripts/campaign/econ/industries/niko_MPC_defenseSatelliteLuddicSuppressor.kt
package data.scripts.campaign.econ.industries

import com.fs.starfarer.api.impl.campaign.econ.impl.BaseIndustry
import com.fs.starfarer.api.impl.campaign.ids.Factions

class niko_MPC_defenseSatelliteLuddicSuppressor : BaseIndustry() {
    override fun apply() {
        super.apply(true)
    }

    override fun getPatherInterest(): Float {
        val ourMarket = getMarket() ?: return 0f
        if (ourMarket.isFreePort) return 0f
        return -3f
    }

    override fun isHidden(): Boolean { //hidden structures dont block structure construction
        return true
    }

    override fun isAvailableToBuild(): Boolean {
        return false
    }

    override fun showWhenUnavailable(): Boolean {
        return false
    }
}